src/main/java/org/hoverla/bibernate/persistence/factory/DefaultSessionFactoryBuilder.java

```
@@ -1,19 +1,41 @@
 package org.hoverla.bibernate.persistence.factory;
 
-import org.hibernate.Session;
-import org.hibernate.internal.SessionImpl;
+import com.zaxxer.hikari.HikariConfig;
+import com.zaxxer.hikari.HikariDataSource;
 import org.hoverla.bibernate.configuration.Configuration;
+import org.hoverla.bibernate.connectionpool.util.BibariDataSource;
 
-public class DefaultSessionFactoryBuilder implements SessionFactoryBuilder {
+import javax.sql.DataSource;
 
-    // private final DataSourceStrategy dataSource;
-    // own or hikari
+public class DefaultSessionFactoryBuilder implements SessionFactoryBuilder {
 
     @Override
     public SessionFactory build(Configuration conf) {
         Configuration.ConnPoolProviderType poolProvider = conf.getPoolProvider();
-        SessionImpl session = null;
-        SessionFactory sessionFactory = new SessionFactoryImpl(null);
-        return sessionFactory;
+        DataSource dataSource = configureDatasourceBasedOnProvider(poolProvider, conf);
+        return new SessionFactoryImpl(dataSource);
+    }
+
+    private DataSource configureDatasourceBasedOnProvider(Configuration.ConnPoolProviderType poolProvider, Configuration conf) {
+        if (poolProvider == Configuration.ConnPoolProviderType.HIKARI) {
+            return configureHikariDataSource(conf);
+        } else if (poolProvider == Configuration.ConnPoolProviderType.BIBARI) {
+            return configureBibariDataSource(conf);
+        }
+        return null;
```

Maybe better throw some `DataSourceNotFoundException` instead of returning `null`